### DISCUSSION TITLE
How does it works offline?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [pieces-app/support#214] [Q&A] [answered]
I want to know how it works being that it says you can run it offline, won't that mean that it will use our CPU/GPU for processing? Does it needs internet access at all? I'm concerned about my privacy here, will I have to send data to anywhere?

**Comment by MovGP0** [comment]
You can install a local LLM for the indexing and on-device data processing/answer generation.

However, most current computers do not meet the hardware requirements.

1. open the `Pieces Desktop` app
2. click on the settings button (the gear icon near bottom/left)
3. Select `Copilot Chats` on the left
4. Click `Manage Copilot LLM Runtime`
5. Cick `On-Device`
6. Select the model you prefer. Notice the `System requirements` of the models.
7. Choose download

**Comment by rosie-at-pieces** [comment, marked as answer]
Just confirming what @MovGP0 said - yes, if you turn off all cloud capabilities you will end up using your computer's resources. You don't need internet access at all to use almost everything in Pieces but if you turn the processing to entirely local there will be some impacts.  @MovGP0 showed you how to use a local model for the copilot, and to use entirely local processing for saving snippets as well go to Settings > Machine Learning > Enrichment Mode to *Local.*

You can see more details about our data and cloud implications on our [Privacy and Security page. ](https://docs.pieces.app/product-highlights-and-benefits/privacy-security-data)
